experimental/packages/opentelemetry-exporter-prometheus/src/export/types.ts

```diff
@@ -56,4 +58,11 @@ export interface ExporterConfig {
    * @default false
    */
   preventServerStart?: boolean;
+
+  /**
+   * HTTP Server where the Prometheus exporter request listener is will be attached.
+   *
+   * @default undefined (create a new server)
+   */
+  server?: Server;
```

Comment:
Note that we already expose the handler https://github.com/open-telemetry/opentelemetry-js/blob/main/experimental/packages/opentelemetry-exporter-prometheus/src/PrometheusExporter.ts#L161 so that people can customize with their own server. So TBO I'd find this unnecessary.

Reply:
Can you elaborate on what are the concerns with allowing the server to be injected here ?

To my outside eyes feels like a missing extension point..
I think even some of the existing tests outline the need.. 
https://github.com/open-telemetry/opentelemetry-js/blob/main/experimental/packages/opentelemetry-exporter-prometheus/test/PrometheusExporter.test.ts#L133
https://github.com/open-telemetry/opentelemetry-js/blob/main/experimental/packages/opentelemetry-exporter-prometheus/test/PrometheusExporter.test.ts#L79



I also thought about introducing a `PrometheusExporterRequestListener`:
To allow something like:
```javascript
const server = http.createServer();
const listener = new PrometheusExporterRequestListener(...);

listener.register(server);
server.listen(...);
```

That might make more clear that this is the intended approach to this use case..
But I'm not sure.. the implementation might end up being the same as the existing PrometheusExporter minus the start/stop server bits..

Reply:
I'm suggesting that you can already create a PrometheusExporter with the constructor option `preventServerStart: true` and take `prometheusExporter.getMetricsRequestHandler()` to insert into your own server implementation, instead of passing the server implementation to the PrometheusExporter as you might already have a router available.

Reply:
Thanks. I got that part.

I'm trying to understand if there are any concerns on with exposing the server like I have in the PR originally or introducing a dedicated request listener class.

Reply:
I may not get your point correctly. Does `prometheusExporter.getMetricsRequestHandler()` serve your requirements? If so, I believe exposing the server is not necessary anyway.

Reply:
@FabioBatSilva I think @legendecas is questioning the introduction any new public interfaces that aren't necessary. not because they create obvious problems, but because they need to be maintained indefinitely and may cause problems in the future that we're not thinking of right now. if your usecase is already reasonably solved by an existing API then it may not be necessary to add another one

Reply:
That is fair..

I think the issue here is that is not very clear that this is the intended solution..
The "getter" feels like a after tough.. 
Improving the docs might have helped but i don't think the API is very intuitive..

I think there is need or a api that exposes the handler without having to  create a server.
PrometheusExporterRequestListener or some way to inject the server.

But let me know.. i'll close the PR if you guys think otherwise..

Reply:
`PrometheusExporter` by its name doesn't impose it to be a Prometheus server. It's the default common setup. And the constructor option `preventServerStart` is supposed to help to avoid creating a server if it is not needed. Like @weyert mentioned at https://github.com/open-telemetry/opentelemetry-js/pull/3481#discussion_r1047930699, we can avoid creating the server handle too if `preventServerStart` is set. 

Improvements are definitely welcomed!

Reply:
Not really sure what implement looks like if exposing the server or extracting the handler are not options..

Reply:
What do you mean? You can already access the request handler? I don't fully understand what the goal of this change is or why it is exactly necessary. Can you give an example how you would use it?

In my code I am doing:
```typescript
this._app.use('/-/metrics', (req: express.Request, res: express.Response, _nextMiddleware: NextFunction) => {
      try {
          const metricReader = this._metricReader
          const isPrometheusExporter = metricReader.constructor.name === 'PrometheusExporter'
          if (isPrometheusExporter) {
            // use any as the metricReader is the API type of MetricReader at this time
            (metricReader as any).getMetricsRequestHandler(req, res)
          } else {
            res.status(500).send('Unexpected exporter, metric reader availalble')
          }
      } catch (err: unknown) {
        res.status(500).json({ status: 'NOT OK', error: (err as Error).message })
      }
    })
```

Reply:
Yes, that will work. 
I get the is possible..

But not a great that you have created a exporter and implement all that just to get access to it..

I'm not sure how I can i make the API better here. Given the feedback I got so far..